### ISSUE TITLE
Missing slash (/) in the beginning of a path

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4380]
|<img alt="alexis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| alexis reported|
|-|-|
|Trac ID|trac#4380|
|Type|defect|
|Created|2010-03-24 16:24:55Z|

When getting a web page that redirects using the "location" header, some sites omit the '/' at the beginning of the location. Web browsers add this / automatically but client.py don't. Here is a small patch to handle these situations.

example urls:
 * http://stopzilla.com/
 * http://linuxcentral.com
 * http://mozilla-europe.org  ( lol ? )

Attachments:

* [client_path_slash_at_beginning.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7f1/7f10f671d10f8076b8dcb0180ffb78d25605fdfe/45adf875c9742b1fe34872afb2ef6ff79cf815f5.diff) (371 bytes) - added by alexis on 2010-03-24 16:25:16Z - 
* [client_301_relative_location.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7f1/7f10f671d10f8076b8dcb0180ffb78d25605fdfe/addb183553f36888d141b4efe378950522cbb188.diff) (893 bytes) - added by alexis on 2010-03-25 09:15:54Z - a better patch

<details><summary>Searchable metadata</summary>

```
trac-id__4380 4380
type__defect defect
reporter__alexis alexis
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords___parse_path_slash _parse path slash
time__1269447895000000 1269447895000000
changetime__1269544837000000 1269544837000000
version__None None
owner__alexis alexis
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to alexis|
|-|-|

Thanks.  Can you add a unit test for this behavior?  Also, it looks like the fix may be a bit more complicated than what's in this patch.  If the request URL was `http://foo.com/bar/baz` and the redirect is to `quux.html`, won't browsers request `http://foo.com/bar/quux`, rather than `http://foo.com/quux.html`?

**Comment by twisted-trac** [comment]
|<img alt="alexis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|alexis commented|
|-|-|

You are right. Here is a better patch that will handle situations like the one you described. I'll try to write unit tests.

**Comment by twisted-trac** [comment]
|<img alt="alexis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|alexis commented|
|-|-|

Oops, it looks like my ticket is a duplicate of [#3384](https://github.com/twisted/twisted/issues/3384).

**Comment by twisted-trac** [comment]
|<img alt="alexis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|alexis commented|
|-|-|

Why is there no twisted/web/test/test_webclient.py ?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

Replying to [alexis](#note_5):
> Oops, it looks like my ticket is a duplicate of [[#3384](https://github.com/twisted/twisted/issues/3384)](https://github.com/twisted/twisted/issues/3384).

Closing it in that case. Can you make a patch against the branch (that also contains other fixes) in #3384? Please reopen if I misunderstood..
